app/models/mixins/retirement_mixin.rb

```diff
@@ -26,6 +27,13 @@ def retire(ids, options = {})
       q_options.merge!(:user_id => user.id, :group_id => user.current_group.id, :tenant_id => user.current_tenant.id) if user
       MiqQueue.put(q_options)
     end
+
+    def set_retirement_requester(obj_name, obj_ids, requester)
+      obj_ids.each do |id|
+        _log.info("Retirement requester for #{obj_name} #{id} being set to #{requester.userid}")
+        obj_name.constantize.find(id).update_attributes(:retirement_requester => requester.userid)
```

Comment:
Have you verified that all object types that pass through this mixin logic have a `retirement_requester`?  They need to be available on `Hammer` and `Gaprindashvili` branches as well.

Reply:
I think that we're at least ```fine```, haha. May 4th of '17 was the date that this field was available on Services n Vms n OrchStacks.